Source/Urho3D/Network/MessageUtils.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2025-2025 the rbfx project.
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT> or the accompanying LICENSE file.
+
+#include "Urho3D/Precompiled.h"
+
+#include "Urho3D/Network/MessageUtils.h"
+
+namespace Urho3D
+{
+
+LargeMessageWriter::LargeMessageWriter(
+    AbstractConnection& connection, NetworkMessageId incompleteMessageId, NetworkMessageId lastMessageId)
+    : connection_(connection)
+    , buffer_(connection_.GetOutgoingMessageBuffer())
+    , debugInfo_(connection_.GetDebugInfoBuffer())
+    , incompleteMessageId_(incompleteMessageId)
+    , lastMessageId_(lastMessageId)
+{
+    buffer_.Clear();
+    debugInfo_.clear();
+}
+
+VectorBuffer& LargeMessageWriter::GetBuffer()
+{
+    return buffer_;
+}
+
+ea::string* LargeMessageWriter::GetDebugInfo()
+{
+#ifdef URHO3D_LOGGING
+    return &debugInfo_;
+#else
+    return nullptr;
+#endif
```

Comment:
Is there a usecase for user modifying this string? If not, this could be `const ea::string&` + `return EMPTY_STRING;` when built without logging.

Reply:
This string is for the user to dump debug info related to the message, so yeah